.circleci/real_config.yml

```diff
@@ -112,85 +112,74 @@ commands:
       - run: git submodule sync
       - run: git submodule update --init
 
-  skip-if-docs-only:
+  add-and-fetch-upstream:
     steps:
       - run: git remote add upstream https://github.com/determined-ai/determined
       - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for docs-only changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = "docs" ] ; then
-              echo "docs-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected non-docs-only changes: $DIFF_DIRS"
-            fi
+
+  skip-if-dir:
+    parameters:
+      dir:
+        type: string
+      not:
```

Comment:
This is in the running for the worst variable name in the history of variables. :stuck_out_tongue: 

Maybe "negate"?

It also appears to not be used anywhere.

Reply:
We're good here.